syntax = "proto3";
package worker;

message TaskRequest {
  string task_id = 1;
  string instruction = 2;
  string context = 3;
  map<string, string> input_data = 4;
}

message TaskResponse {
  string task_id = 1;
  string status = 2; // "success", "error"
  string result = 3;
  string error_message = 4;
}

service WorkerService {
  rpc ExecuteTask (TaskRequest) returns (TaskResponse);
}
